src/coreclr/tools/Common/JitInterface/LoongArch64PassStructInRegister.cs

```diff
@@ -0,0 +1,168 @@
+            // and the user can always change their structure to something more descriptive of what they want
+            // instead of adding additional padding at the end of a one-field structure.
+            // We do this check here to save looking up the FixedBufferAttribute when loading the field
+            // from metadata.
+            bool isFixedBuffer = numIntroducedFields == 1
+                                    && firstFieldElementType.IsValueType
+                                    && firstField.Offset.AsInt == 0
+                                    && mdType.HasLayout()
+                                    && ((typeDesc.GetElementSize().AsInt % firstFieldSize) == 0);
+
+            if (isFixedBuffer)
+            {
+                numIntroducedFields = typeDesc.GetElementSize().AsInt / firstFieldSize;
+                if (numIntroducedFields > 2)
+                {
+                    return (uint)LAFlags.STRUCT_NO_FLOAT_FIELD;
+                }
+            }
+
+            int fieldIndex = 0;
+            foreach (FieldDesc field in typeDesc.GetFields())
+            {
+                if (fieldIndex > 1)
+                {
+                    return (uint)LAFlags.STRUCT_NO_FLOAT_FIELD;
+                }
+
+                Debug.Assert(fieldIndex < numIntroducedFields);
+
+                switch (field.FieldType.Category)
+                {
+                    case TypeFlags.Double:
+                    {
+                        if (numIntroducedFields == 1)
+                            floatFieldFlags = (uint)LAFlags.STRUCT_FLOAT_FIELD_ONLY_ONE;
+                        else if (fieldIndex == 0)
+                            floatFieldFlags = (uint)LAFlags.STRUCT_FIRST_FIELD_DOUBLE;
+                        else if ((floatFieldFlags & (uint)LAFlags.STRUCT_FLOAT_FIELD_FIRST) != 0)
+                            floatFieldFlags = floatFieldFlags ^ (uint)LAFlags.STRUCT_MERGE_FIRST_SECOND_8;
+                        else
+                            floatFieldFlags |= (uint)LAFlags.STRUCT_SECOND_FIELD_DOUBLE;
+                    }
+                    break;
+
+                    case  TypeFlags.Single:
+                    {
+                        if (numIntroducedFields == 1)
+                            floatFieldFlags = (uint)LAFlags.STRUCT_FLOAT_FIELD_ONLY_ONE;
+                        else if (fieldIndex == 0)
+                            floatFieldFlags = (uint)LAFlags.STRUCT_FLOAT_FIELD_FIRST;
+                        else if ((floatFieldFlags & (uint)LAFlags.STRUCT_FLOAT_FIELD_FIRST) != 0)
+                            floatFieldFlags ^= (uint)LAFlags.STRUCT_MERGE_FIRST_SECOND;
+                        else
+                            floatFieldFlags |= (uint)LAFlags.STRUCT_FLOAT_FIELD_SECOND;
+                    }
+                    break;
+
+                    case TypeFlags.ValueType:
+                    //case TypeFlags.Class:
```

Comment:
Hi, @jkotas 
I need your help, I don't know how to get the field type for `case TypeFlags.Class:`.

Reply:
I am not sure what you mean. `field.FieldType` is the actual type of the field, but I do not think you actually need it for this case.

You should handle this case the same way as `ELEMENT_TYPE_CLASS` case in the VM implementation.